core/lib/server/pub.ts

```diff
@@ -1537,9 +1554,9 @@ function nestRelatedPubsAndChildren<Options extends GetPubsWithRelatedValuesAndC
 	const processedPubsById = new Map<PubsId, ProcessedPub<Options>>();
 
 	function processPub(pubId: PubsId, depth: number): ProcessedPub<Options> | undefined {
-		// if (depth < 0) {
-		// 	return processedPubsById.get(pubId);
-		// }
+		if (depth < 0) {
+			return processedPubsById.get(pubId);
+		}
```

Comment:
Uncommenting this fixed an infinite recursion issue when selecting pubs with a cycle